src/main.rs

```diff
@@ -1,27 +1,56 @@
-use std::hash::Hash;
-use std::io::ErrorKind;
-use std::iter::FromIterator;
-use std::{
-    collections::HashSet,
-    fs::{self, File},
-    process::Command,
-};
-
+mod handler;
 use anyhow::{bail, Error, Result};
 use clap::{Parser, Subcommand, ValueEnum};
+use config::{Config, File, FileFormat};
 use glob::glob;
-use serde::{Deserialize, Serialize};
+use handler::*;
+use serde::Deserialize;
+use std::path::Path;
+use std::process::Command;
+use std::str;
+
+static GREENBOOT_INSTALL_PATHS: [&str; 2] = ["/usr/lib/greenboot", "/etc/greenboot"];
+static GREENBOOT_CONFIG_FILE: &str = "/etc/greenboot/greenboot.conf";
 
 #[derive(Parser)]
 #[clap(author, version, about, long_about = None)]
 #[clap(propagate_version = true)]
 struct Cli {
     #[clap(value_enum, short, long, default_value_t = LogLevel::Info)]
     log_level: LogLevel,
-
     #[clap(subcommand)]
     command: Commands,
 }
+#[derive(Debug, Deserialize)]
+struct GreenbootConfig {
+    //max reboot attempts if diagnostics fails
+    max_reboot: i32,
+}
+
+impl GreenbootConfig {
+    fn set_default() -> Self {
+        Self { max_reboot: 3 }
+    }
+
+    fn get_config() -> Self {
+        let mut config = Self::set_default();
+        let parsed =
+            Config::builder().add_source(File::new(GREENBOOT_CONFIG_FILE, FileFormat::Ini));
+        match parsed.build() {
+            Ok(c) => {
+                config.max_reboot = match c.get_int("GREENBOOT_MAX_BOOT_ATTEMPTS") {
+                    Ok(c) => c.try_into().unwrap(),
```

Comment:
We cannot use an unchecked `unwrap`, it might panic.
Also, what if this `c` is negative?

Reply:
since the variable is a signed int a negative value will not have any impact on the code itself.
Also putting a - value  allows users some level ability ability by blocking the system reboot even if the health-check fails post upgrade.

Reply:
fixed the unwrap()

Reply:
OK, if the negative numbers are allowed I would simply log a warning for the user to see that the reboots won't be effective (unless they use the `trigger_rollback` via CLI)